website/docs/configuration/expressions/splat.html.md

```diff
@@ -39,34 +39,75 @@ The above expression is equivalent to the following `for` expression:
 [for o in var.list : o.interfaces[0].name]
 ```
 
-Splat expressions are for lists only (and thus cannot be used [to reference resources
-created with `for_each`](/docs/configuration/meta-arguments/for_each.html#referring-to-instances),
-which are represented as maps in Terraform). However, if a splat expression is applied
-to a value that is _not_ a list or tuple then the value is automatically wrapped in
-a single-element list before processing.
+## Splat Expressions with Maps
 
-For example, `var.single_object[*].id` is equivalent to `[var.single_object][*].id`,
-or effectively `[var.single_object.id]`. This behavior is not interesting in most cases,
-but it is particularly useful when referring to resources that may or may
-not have `count` set, and thus may or may not produce a tuple value:
+Splat expressions have their iteration effect for lists, sets, and tuples only.
```

Comment:
> have their iteration effect

I find that phrase awkward to parse, can it be simplified a bit? `Splat expressions work with` or `Splat expressions can iterate lists...`, something like those?

Reply:
I did struggle here because "This only works with lists" didn't feel clear here... it seemed to me like it was suggesting that using other types would return an _error_, which it doesn't.

However, I know that "their iteration effect" is far from ideal. I think I'll revise this to talk about it in the opposite sense: "Splat expressions work differently for value types other than lists, sets and tuples.". I think that gets the point across more directly, because for the purpose of this section it really doesn't matter what splat expressions do with lists, sets, and tuples.